pkg/controller/jobframework/interface.go

```diff
@@ -51,6 +51,9 @@ type GenericJob interface {
 	PodsReady() bool
 	// GetGVK returns GVK (Group Version Kind) for the job.
 	GetGVK() schema.GroupVersionKind
+}
+
+type JobWithReclaimablePods interface {
```

Comment:
Do you plan to add more functions to this interface? If not, adding this interface only for `ReclaimablePods` might not be worth it.

Reply:
I agree with Yuki here. Returning `nil` seems simple enough.
If we split interfaces, a CRD that can implement reclaimable pods needs to make sure they implement each interface correctly, without typos. Having one interface makes the problem smaller.

Reply:
This should be all for Reclaimable. The idea is that an integration that is planing to support partial admission should not be required to implement the empty method. 



> I agree with Yuki here. Returning `nil` seems simple enough.

It's fine when we only have one method, but even at this point is not the case, `UpdateStatus` should also be optional since is actually useful foe batch/job only.

Reply:
> It's fine when we only have one method, but even at this point is not the case, UpdateStatus should also be optional since is actually useful foe batch/job only.

I'm fine.